From 9e812be46b6f7be61524e14652ba380c02518262 Mon Sep 17 00:00:00 2001
From: "Tadashi G. Takaoka" <takaoka@google.com>
Date: Fri, 16 Sep 2016 12:12:14 +0900
Subject: [PATCH] DO NOT MERGE: Catch all exceptions when parsing IME meta data

Bug: 30568284
Change-Id: I0b613f8ce0f014320c5ac1bf445699ea2702a0a2
(manually cherry picked from 9b2997d22e6ce2a15065d8e7608dd77b316c2065)

(cherry picked from commit 7bb4c825585d5a6afbb59484ec7db5f5dec1bdaf)
---
 .../core/java/com/android/server/InputMethodManagerService.java     | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/services/core/java/com/android/server/InputMethodManagerService.java b/services/core/java/com/android/server/InputMethodManagerService.java
index 6d07a57..f2058a5 100644
--- a/services/core/java/com/android/server/InputMethodManagerService.java
+++ b/services/core/java/com/android/server/InputMethodManagerService.java
@@ -2949,10 +2949,8 @@ void buildInputMethodListLocked(ArrayList<InputMethodInfo> list,
                     Slog.d(TAG, "Found an input method " + p);
                 }
 
-            } catch (XmlPullParserException e) {
-                Slog.w(TAG, "Unable to load input method " + compName, e);
-            } catch (IOException e) {
-                Slog.w(TAG, "Unable to load input method " + compName, e);
+            } catch (Exception e) {
+                Slog.wtf(TAG, "Unable to load input method " + compName, e);
             }
         }
 
